plugins/factory-plugin/src/workspace-projects-manager.ts

```diff
@@ -57,4 +63,63 @@ export class WorkspaceProjectsManager {
         theia.window.showInformationMessage("Che Workspace: Finished cloning projects.");
     }
 
+    async syncWorkspace() {
+        const gitConfigPattern = '**/.git/{HEAD,config}';
+        const gitConfigWatcher = theia.workspace.createFileSystemWatcher(gitConfigPattern);
```

Comment:
` FileSystemWatcher extends Disposable`, so clean up, please, file watcher resources. You could use context :
https://github.com/eclipse/che-theia/blob/master/plugins/containers-plugin/src/containers-plugin.ts#L15
https://github.com/eclipse/che-theia/blob/master/plugins/containers-plugin/src/containers-plugin.ts#L20